### PR TITLE
Merge to ecf baseline

### PR DESCRIPTION
**Pull request by romoh**
From Chri's BareMetal2 to ecf.

Validated it builds.

### CODE REVIEW

File: api/v1alpha3/azurestackhcivirtualmachine_types.go [hunk 1]
```diff
@@ -71,6 +74,12 @@ type AzureStackHCIVirtualMachineStatus struct {
 	// Conditions defines current service state of the AzureStackHCIVirtualMachine.
 	// +optional
 	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+
+	// CloudResourceName is the name resource used by the machine.
+	// For VMs, this will be the same as Name. For bare-metal machines, it will be the name
+	// of the host.
+	// +optional
+	CloudResourceName string `json:"cloudResourceName,omitempty"`
```
Comment (jiria) [inline api/v1alpha3/azurestackhcivirtualmachine_types.go:82]: Did we miss this one?

File: controllers/azurestackhcicluster_controller.go [hunk 2]
```diff
@@ -146,6 +146,14 @@ func (r *AzureStackHCIClusterReconciler) reconcileNormal(clusterScope *scope.Clu
 		return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 20}, nil
 	}
 
+	if clusterScope.AzureStackHCICluster.Name != "clustergroup-management" {
```
Comment (jiria) [inline controllers/azurestackhcicluster_controller.go:149]: We should split this one into a separate hack PR.

File: cloud/services/virtualmachines/virtualmachines.go [hunk 3]
```diff
@@ -213,6 +288,15 @@ func (s *Service) Delete(ctx context.Context, spec interface{}) error {
 	return err
 }
 
+func (s *Service) clearBareMetalMachine(ctx context.Context, vmSpec *Spec) error {
```
Comment (cwize1) [inline cloud/services/virtualmachines/virtualmachines.go:291]: We probably don't need this function anymore.

**Review by cwize1 — APPROVED** [review]

**Comment by nwoodmsft** [comment]
Master has opened up this week. Are you ready for review or still iterating on the PR?

**Pull request merged**